src/components/Error/ErrorBoundary.tsx

```diff
@@ -54,11 +60,24 @@ export default class ErrorBoundary extends Component<PropsWithChildren<ErrorBoun
   }
 
   renderErrorDialog(error: Error) {
-    const header = error instanceof WidgetError ? error.header : DEFAULT_ERROR_HEADER
-    const action = error instanceof WidgetError ? error.action : DEFAULT_ERROR_ACTION
+    const header =
+      error instanceof WidgetError ? (error.dismissable ? DISMISS_ERROR_HEADER : error.header) : DEFAULT_ERROR_HEADER
```

Comment:
I don't think you should be overwriting the header here:

    const isDismissable = error instanceof WidgetError && error.dismissable
    const header = error instanceof WidgetError ? (error.header) : (isDismissable ? DISMISS_ERROR_HEADER : DEFAULT_ERROR_HEADER)

Reply:
actually i think this version is has a problem too: we only set the `dismissable` attribute on `WidgetErrors` , so in this case we'd never actually show `DISMISS_ERROR_HEADER`

i made an update that should address this in a different way (setting the attributes as the defaults for `SwapError`s)